### PR TITLE
feat: add new MSK public_fulfilment_topic to dev

### PR DESCRIPTION
**Pull request by jturner-uw**

### CODE REVIEW

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: d82d5bd4e538c56d229b43b4717a6901436afcb4
Requested At: 2024-09-06T09:59:03Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: d82d5bd4e538c56d229b43b4717a6901436afcb4
Requested At: 2024-09-06T09:59:03Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: 376ff6e368cd1ae69ca71aeb3e566b716731a313
Requested At: 2024-09-06T10:07:58Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

File: dev-aws/kafka-shared-msk/customer-billing/topics.tf [hunk 1]
```diff
@@ -116,3 +116,20 @@ resource "kafka_topic" "bex_legacy_invoice_api" {
...
 }
+
+resource "kafka_topic" "public_fulfilment_events" {
+  name               = "bex.public.fulfilment_events"
+  replication_factor = 3
+  partitions         = 10
+  config = {
+    "remote.storage.enable" = "true"
+    "retention.bytes"       = "-1"
+    "retention.ms"          = "-1"
```
Comment (sbuliarca) [inline dev-aws/kafka-shared-msk/customer-billing/topics.tf:127]: Do you need infinite retention in dev?

Comment (jturner-uw) [reply dev-aws/kafka-shared-msk/customer-billing/topics.tf:127]: good catch, no we do not need this in dev.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: fbfd4b541ec6cfa1c4531d2cc4801d9400332770
Requested At: 2024-09-06T12:44:27Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

File: dev-aws/kafka-shared-msk/customer-billing/topics.tf (hunk 1 above)
Comment (jturner-uw) [reply dev-aws/kafka-shared-msk/customer-billing/topics.tf]: Sorry, hadn't pushed the changes :man_facepalming:

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**